packages/material-ui-lab/src/Masonry/Masonry.test.js

```diff
@@ -0,0 +1,125 @@
+import { expect } from 'chai';
+import * as React from 'react';
+import { createClientRender, describeConformanceV5 } from 'test/utils';
+import Masonry, { masonryClasses as classes } from '@material-ui/lab/Masonry';
+
+const itemsData = [
+  {
+    img: '/fake1.png',
+    title: 'fake1',
+  },
+  {
+    img: '/fake2.png',
+    title: 'fake2',
+  },
+];
+
+describe('<Masonry />', () => {
+  const render = createClientRender();
+
+  describeConformanceV5(
+    <Masonry>
+      <div />
+    </Masonry>,
+    () => ({
+      classes,
+      inheritComponent: 'div',
+      render,
+      refInstanceof: window.HTMLDivElement,
+      testComponentPropWith: 'div',
+      muiName: 'MuiMasonry',
+      skip: ['componentsProp'],
+    }),
+  );
+
+  const children = itemsData.map((item, idx) => (
+    <div key={idx} data-testid="test-children">
+      <img src={item.img} alt={item.title} />
+    </div>
+  ));
+
+  it('should render with the root class by default', () => {
+    const { getByTestId } = render(<Masonry data-testid="test-root">{children}</Masonry>);
+    expect(getByTestId('test-root')).to.have.class(classes.root);
+  });
+
+  it('should render children by default', () => {
+    const { getAllByTestId } = render(<Masonry>{children}</Masonry>);
+    expect(getAllByTestId('test-children').length).to.equal(2);
+  });
+
+  describe('style attribute:', () => {
+    it('should overwrite style', () => {
+      const style = { backgroundColor: 'black' };
+      const { getByTestId } = render(
+        <Masonry style={style} data-testid="test-root">
+          {children}
+        </Masonry>,
+      );
+
+      expect(getByTestId('test-root')).toHaveInlineStyle({ backgroundColor: 'black' });
+    });
+
+    it('should render with a grid-auto-rows of 0 by default', () => {
+      const { getByTestId } = render(<Masonry data-testid="test-root">{children}</Masonry>);
+      const style = getComputedStyle(getByTestId('test-root'));
+      expect(style['grid-auto-rows']).to.equal('0');
+    });
+  });
+
+  describe('props:', () => {
+    describe('prop: component', () => {
+      it('should render a div by default', () => {
+        const { container } = render(<Masonry>{children}</Masonry>);
+        expect(container.firstChild).to.have.property('nodeName', 'DIV');
+      });
+
+      it('should render a different component', () => {
+        const { container } = render(<Masonry component="span">{children}</Masonry>);
+        expect(container.firstChild).to.have.property('nodeName', 'SPAN');
+      });
+    });
+
+    describe('prop: className', () => {
+      it('should append the className to the root element', () => {
+        const { container } = render(<Masonry className="foo">{children}</Masonry>);
+        expect(container.firstChild).to.have.class(classes.root);
+        expect(container.firstChild).to.have.class('foo');
+      });
+    });
+
+    describe('prop: cols', () => {
```

Comment:
If we want to test the styles, I would suggest doing what's have in the `Stack` component (exporting the style and testing the results there). Otherwise, you will need to take care of adjusting the expected CSS to reflect how it is done in the browsers, for example see the failing tests https://app.circleci.com/pipelines/github/mui-org/material-ui/49474/workflows/46624040-1700-48b0-b053-e698a9bdbbe0/jobs/276216